### PR TITLE
8252387: Deprecate for removal css Selector and ShapeConverter constructors

### PR DESCRIPTION
**Pull request by bhaweshkc**
Deprecate the public constructor of javafx.css.Selector as it should not be public due to only being extended by classes in same package.
Deprecate the public constructor of javafx.css.converter.ShapeConverter as its a singleton class.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252387](https://bugs.openjdk.java.net/browse/JDK-8252387): Deprecate for removal css Selector and ShapeConverter constructors

### Reviewers
 * [Nir Lisker](https://openjdk.java.net/census#nlisker) (@nlisker - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/290/head:pull/290`
`$ git checkout pull/290`

### CODE REVIEW

**Comment by kevinrushforth** [comment]
> Constrcutor for class javafx.css.Selector should not be public as it is only extended by classes in same package. it should be changed to package-private
> Constrcutor for class javafx.css.converter.ShapeConverter should be private rather than public as its a singleton class.

This is the eventual goal (for JavaFX 17), but it isn't what this bug is addressing for JavaFX 16. Can you reword this to indicate that this bug is just deprecating the constructors for removal, and that we will file a follow-up bug to later change the access?

File: modules/javafx.graphics/src/main/java/javafx/css/Selector.java [hunk 1]
```diff
@@ -42,6 +42,13 @@
  */
 abstract public class Selector {
 
+    /**
+     * @deprecated This constructor was exposed erroneously and will be removed in the next version.
+     */
```
Comment (nlisker) [inline modules/javafx.graphics/src/main/java/javafx/css/Selector.java:47]: Please direct the reader to the way of obtaining an instance of this class like you did for `ShapeConverter`.

**Review by nlisker — COMMENTED** [review]
Looks good. Added a single comment.

File: modules/javafx.graphics/src/main/java/javafx/css/Selector.java (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/javafx/css/Selector.java:47]: Probably you can refer to the `createSelector` method.

**Review by nlisker — APPROVED** [review]

**Comment by kevinrushforth** [comment]
This should not have been marked as ready by the Skara bot. Looks like there is a bug in the processing of the `/csr` since the CSR is not yet approved.

@bhaweshkc please wait until the CSR is approved before integrating.

**Comment by nlisker** [comment]
@kevinrushforth Why is the Skara bot saying this PR can be integrated if the CSR is not approved yet?

Also, in the main post under Reviewers, the links to the users lead to a 404.

Should I submit bugs for these?

**Comment by kevinrushforth** [comment]
> Why is the Skara bot saying this PR can be integrated if the CSR is not approved yet?

Looks like we both noticed this at the same time.

> Also, in the main post under Reviewers, the links to the users lead to a 404.

I hadn't noticed this before.

> Should I submit bugs for these?

Yes, please.

**Comment by kevinrushforth** [comment]
@bhaweshkc the CSR is approved. Go ahead and `/integrate` when ready.
